src/ompl/geometric/planners/bitstar/BITstar.h

```diff
@@ -0,0 +1,614 @@
+            as the raw data. (numBatches_) */
+            unsigned int numBatches() const;
+            /** \brief Retrieve the number of batches processed
+            as a planner-progress property. (numBatches_) */
+            std::string batchesProgressProperty() const;
+
+            /** \brief Retrieve the number of graph prunings performed
+            as a planner-progress property. (numPrunings_) */
+            std::string pruningProgressProperty() const;
+
+            /** \brief Retrieve the \e total number of states generated
+            as a planner-progress property. (numSamples_) */
+            virtual std::string totalStatesCreatedProgressProperty() const;
+
+            /** \brief Retrieve the \e total number of vertices added to the graph
+            as a planner-progress property. (numVertices_) */
+            std::string verticesConstructedProgressProperty() const;
+
+            /** \brief Retrieve the number of states pruned from the problem
+            as a planner-progress property. (numFreeStatesPruned_) */
+            std::string statesPrunedProgressProperty() const;
+
+            /** \brief Retrieve the number of graph vertices that are disconnected and
+            either returned to the set of free samples or deleted completely
+            as a planner-progress property. (numVerticesDisconnected_) */
+            std::string verticesDisconnectedProgressProperty() const;
+
+            /** \brief Retrieve the number of global-search edges that rewired the graph
+            as a planner-progress property. (numRewirings_) */
+            std::string rewiringProgressProperty() const;
+
+            /** \brief Retrieve the number of state collisions checks (i.e., calls to SpaceInformation::isValid(...))
+            as a planner-progress property. (numStateCollisionChecks_) */
+            std::string stateCollisionCheckProgressProperty() const;
+
+            /** \brief Retrieve the number of edge (or motion) collision checks (i.e., calls to SpaceInformation::checkMotion(...))
+            as a planner-progress property. (numEdgeCollisionChecks_) */
+            std::string edgeCollisionCheckProgressProperty() const;
+
+            /** \brief Retrieve the number of nearest neighbour calls (i.e., NearestNeighbors<T>::nearestK(...) or NearestNeighbors<T>::nearestR(...))
+            as a planner-progress property. (numNearestNeighbours_) */
+            std::string nearestNeighbourProgressProperty() const;
+            ///////////////////////////////////////
+
+        protected:
+            //Everything is only protected so we can create modifications without duplicating code by deriving from the class:
+
+            //Typedefs:
+            /** \brief A pair of vertices, i.e., an edge. */
+            typedef std::pair<VertexPtr, VertexPtr> vertex_pair_t;
+
+            /** \brief The OMPL::NearestNeighbors structure. */
+            typedef boost::shared_ptr< NearestNeighbors<VertexPtr> > vertex_nn_ptr_t;
```

Comment:
The type name used here is not consistent with other types used above. Please use the Ptr suffix and a CamelCase name